### PR TITLE
Consolidating all JVM access to RuntimeBehavior

### PR DESCRIPTION
**Pull request by krmahadevan**
Fixes # .

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

**Review by juherr — COMMENTED** [review]
The idea is good but I think we should split the configuration of the internal package and the configuration of reporters (and keep specific reporter configuration and its dedicated class).

**Comment by krmahadevan** [comment]
@juherr

Any detailed pointers on how we can go about what you are suggesting ? That
would help me improvise and wrap this PR to closure.

My intent behind this PR was pretty straightforward. First consolidate all
the JVM arguments that TestNG uses, into one single place, so that if
needed we can just route our users to the documentation of this class for
understanding the JVM arguments and its effects.

Since these JVM arguments already exist and are meant to alter behavior
(they are just not documented properly), I thought it was better if they
all resided in one single place.

**Comment by juherr** [comment]
In fact, I see 2 problems:
- dependency from the public packages to the internal package
- hard link between reporter and core which are 2 different things (IMO, it should be in separate modules)

**Comment by juherr** [comment]
Just imagine all reporters are in another module (because of dependencies or packaging reason) and will just be integrated into an uber jar.

From a dependency point of view, it should be
API (public) <- Core (internal)
             <- Reporter (default)

**Comment by krmahadevan** [comment]
@juherr - I am not familiar with the module features of JDK9. So I cant comment and that is perhaps why I still am not able to get the gist of what you are suggesting. The reporters that we are currently dealing with all are part of the core codebase. Not sure why they would all move into a different module. 

Either ways, I would like to wrap up this PR, so please let me know what I should be doing [ I will try to read up about [project jigsaw](http://www.baeldung.com/project-jigsaw-java-modularity) to get a better understanding.]

From your comments I am guessing that the gist is as below:

* Create a class similar to `RuntimeBehavior`, house it within `org.testng.reporters` package and move the below newly added methods into it
  * `org.testng.internal.RuntimeBehavior#verboseMode`
  * `org.testng.internal.RuntimeBehavior#getDefaultEmailableReport2Name`
  * `org.testng.internal.RuntimeBehavior#getDefaultEmailableReportName`
  * `org.testng.internal.RuntimeBehavior#getDefaultStacktraceLevels`
  * `org.testng.internal.RuntimeBehavior#getDefaultFileNameForXmlReports`
  * `org.testng.internal.RuntimeBehavior#getDefaultLineSeparator`
  * `org.testng.internal.RuntimeBehavior#getLineSeparatorOrNewLine`

Is my understanding correct? If you have anything to add, please let me know.

**Comment by juherr** [comment]
In fact, it is not related to Java9 modules but Maven ones (I think it is called projects with gradle) where modules only declare dependencies they are using (ie: no yaml in core could be possible).

A dedicated RuntimeBehavior for reporters is better even if we can see each reporter as a single class with no dependency on others.

File: src/main/java/org/testng/TestNG.java [hunk 1]
```diff
@@ -113,7 +114,7 @@
 
   /** System properties */
   public static final String SHOW_TESTNG_STACK_FRAMES = "testng.show.stack.frames";
```
Comment (juherr) [inline src/main/java/org/testng/TestNG.java]: Why keeping this one here?

**Review by juherr — APPROVED** [review]

File: src/main/java/org/testng/TestNG.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/TestNG.java]: Thanks for pointing that one out. It was a miss. I just grepped the codebase for `System.getProperty()` usages. Fixed this one as well.

**Comment by sskorol** [comment]
@krmahadevan @juherr just wondering if you want to take a look at http://owner.aeonbits.org?
It's a very powerful and flexible library for working with any kind of props, env vars, etc. It even has a hot reload mechanism, which could be very useful in some cases.

**Comment by juherr** [comment]
@sskorol It looks awesome but the current @cbeust's strategy for dependencies is "the fewer the better"

**Review by juherr — APPROVED** [review]

**Comment by krmahadevan** [comment]
@sskorol - Thanks for sharing that library. As @juherr also pointed out,  I have seen that @cbeust is hesitant in terms of adding more libraries to the codebase unless and until its absolutely required.

**Comment by sskorol** [comment]
Ok, I see :)

**Review by cbeust — APPROVED** [review]

**Pull request merged**